lib/eventlog/src/evtstr.c

```diff
@@ -117,7 +117,7 @@ evt_str_append_escape_bs(EVTSTR *es,
           escaped_buffer[escaped_buffer_length++] = unescaped[i];
         }
 
-      if (escaped_buffer_capacity < escaped_buffer_length + escaped_char_max_len)
+      if (escaped_buffer_capacity <= escaped_buffer_length + escaped_char_max_len)
```

Comment:
Nice catch.
The buffer overwrite could happen because of `sprintf` writing a fifth byte to the end of the 4-byte sequence.
I think in this case, it would be cleaner to display that +1 byte explicitly in the condition or we should at least mention it in the commit message.

Reply:
sure I could mention in the comment, but would rather avoid +1ing conditions.

Reply:
@Kokan I think I understand your point: you don't want to highlight sprintf() implementation detail (that it actually copies n+1 bytes) as the if statement would become a less clear code.